src/common/hooks/useShouldZeroApprove/useShouldZeroApprove.ts

```diff
@@ -0,0 +1,33 @@
+import { useApprovalStateForSpender } from 'lib/hooks/useApproval'
+import { useTradeSpenderAddress } from '../useTradeSpenderAddress'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+import { useEffect, useState } from 'react'
+import { useTokenContract } from 'legacy/hooks/useContract'
+import { shouldZeroApprove as shouldZeroApproveFn } from './shouldZeroApprove'
+
+// TODO: Handle tokens that don't allow approvals larger than the balance of the wallet
```

Comment:
This was something mentioned by @shoom3301 as a potential edge case where some tokens don't allow you to do infinite approvals, but only maximum amount you have at that time in your wallet.

Since it needs some research, left it here to at least remember it for all of us.

Reply:
I think we already have that for regular approvals https://github.com/cowprotocol/cowswap/blob/57d43e2fb5e40bf1695cb6b8c7037cfdbd351640/src/common/hooks/useApproveCallback.ts#L27